src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -466,4 +478,24 @@ QgsPointCloudBlock *QgsEptDecoder::decompressLaz( const QString &filename,
          );
 }
 
+QgsPointCloudBlock *QgsEptDecoder::decompressLaz( const QByteArray &byteArrayData,
+    const QString &nodeStr,
+    const QgsPointCloudAttributeCollection &attributes,
+    const QgsPointCloudAttributeCollection &requestedAttributes )
+{
+  QString filename = QDir::tempPath() + QDir::separator() + QStringLiteral( "%1.laz" ).arg( nodeStr );
+  std::ofstream file( filename.toStdString(), std::ios::binary | std::ios::out );
+  if ( file.is_open() )
+  {
+    file.write( byteArrayData.constData(), byteArrayData.size() );
+    file.close();
+  }
+  else
+  {
+    QgsDebugMsg( QStringLiteral( "Couldn't open %1" ).arg( filename ) );
+    return nullptr;
+  }
+  return QgsEptDecoder::decompressLaz( filename, attributes, requestedAttributes );
+}
```

Comment:
We are not removing the temporary files after they are not needed anymore... my temp folder is filling up with .laz files :smile: